File: leetcode/weekcontest/2022-7-17/1.go
package _022_7_17

func numberOfPairs(nums []int) []int {
	mp := map[int]int{}
	for i := range nums {
		mp[nums[i]]++
	}
	cnt := 0
	for k, _ := range mp {
		cnt += mp[k] / 2
	}
	return []int{cnt, len(nums) - cnt*2}
}

File: leetcode/weekcontest/351/1.py
from math import gcd

class Solution:
    def countBeautifulPairs2(self, nums: List[int]) -> int:
        ans = 0
        n = len(nums)
        for i in range(n):
            x = nums[i]
            while x >= 10: x //= 10
            for j in range(i+1, n):
                last = nums[j] % 10
                if gcd(last, x) == 1:
                    ans += 1
        return ans
    
    def countBeautifulPairs(self, nums: List[int]) -> int:
        ans = 0
        n = len(nums)
        cnt = [0] * n
        for x in nums:
            for j in range(1, 10):
                if gcd(x % 10, j) == 1:
                    ans += cnt[j]
            while x >= 10: x //= 10
            cnt[x] += 1
        return ans  
        
